### PR TITLE
Multiple Verifier Improvements

### PR DESCRIPTION
**Pull request by caithagoras**
```
== RELEASE NOTES ==

Verifier Changes
* Add support for overriding session properties for all queries in a suite.
* Add cluster type and retryable information for ``QueryFailure``.
* Add final query failure information to Verifier output event.
```

### CODE REVIEW

**Comment by mayankgarg1990** [comment]
Re: the change log - 


> * Add configuration properties ``control.session-properties-override-strategy`` and ``control.session-properties-override`` to support overriding session properties for all queries in the suite.......

We already have a wiki with information about verifier, lets add the details there. We should not add so much details to release notes.

> * Add final query failure to Verifier output event.

> * Add new values ``CONTROL_SETUP``, ``CONTROL_MAIN``, ``CONTROL_TEARDOWN``,
  ``TEST_SETUP``, ``TEST_MAIN``, ``TEST_TEARDOWN``, ``DETERMINISM_ANALYSIS``
  to ``QueryStage``, to indicate the nature of queries that are failed.

Should we just write something like (just giving a sense) `Add `query..type` and `final failure` information to the output event logs

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/QueryFailure.java [hunk 1]
```diff
@@ -27,23 +28,32 @@
...
 public class QueryFailure
 {
+    private final String cluster;
     private final String queryStage;
     private final String errorCode;
     private final String prestoQueryId;
     private final String stacktrace;
 
     public QueryFailure(
+            ClusterType cluster,
```
Comment (rongrong) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/event/QueryFailure.java]: I'd still name the variable `clusterType`. `cluster` can mean different things.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/QueryFailure.java [hunk 2]
```diff
@@ -27,23 +28,32 @@
...
-        this.queryStage = queryStage.name();
+        this.cluster = requireNonNull(cluster.name(), "cluster is null");
+        this.queryStage = requireNonNull(queryStage.name(), "queryStage is null");
         this.errorCode = requireNonNull(errorCode, "errorCode is null");
         this.prestoQueryId = prestoQueryId.orElse(null);
         this.stacktrace = requireNonNull(stacktrace, "stacktrace is null");
     }
 
+    @EventField
+    public String getCluster()
```
Comment (rongrong) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/event/QueryFailure.java]: Same here, use `getClusterType`.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/JdbcPrestoAction.java [hunk 3]
```diff
@@ -182,7 +182,7 @@ private PrestoConnection getConnection(
     {
         PrestoConnection connection = DriverManager.getConnection(
                 clusterUrls.get(queryOrigin.getCluster()),
-                configuration.getUsername(),
+                configuration.getUsername().orElse(null),
```
Comment (rongrong) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/JdbcPrestoAction.java]: What's the motivation to allow absent username?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java [hunk 4]
```diff
@@ -13,15 +13,27 @@
...
 package com.facebook.presto.verifier.framework;
 
+import java.util.Map;
 import java.util.Optional;
 
 public interface QueryConfigurationOverrides
 {
+    enum SessionPropertiesOverrideStrategy
+    {
+        NO_OVERRIDE,
```
Comment (rongrong) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfigurationOverrides.java]: Theoretically `SUBSTITUTE` is also `NO_OVERRIDE`. Maybe something like `NO_ACTION`?

**Comment by caithagoras** [comment]
@mayankgarg1990 release notes updated
@rongrong comments addressed

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/JdbcPrestoAction.java (hunk 3 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/JdbcPrestoAction.java]: ``java.sql.DriverManager`` treats `user` and `password` equally, as nullable parameters, so we don't need to arbitrarily put a restriction on `user`.

**Comment by rongrong** [comment]
Please fix the build.

**Pull request merged**